File: evals/refs/rev.py
import sys
lines = sys.stdin.read().split('\n')
i = 0
T = int(lines[i]); i += 1
out = []
for _ in range(T):
    N = int(lines[i]); i += 1
    nums_line = lines[i]; i += 1
    if N == 0:
        out.append('')
        continue
    nums = nums_line.split()
    result = []
    k = len(nums) - 1
    while k >= 0:
        result.append(nums[k])
        k -= 1
    out.append(' '.join(result))
sys.stdout.write('\n'.join(out) + '\n')

File: evals/refs/rev.rs
use std::io::{self, BufRead, Write, BufWriter};

fn main() {
    let stdin = io::stdin();
    let stdout = io::stdout();
    let mut out = BufWriter::new(stdout.lock());
    let mut lines = stdin.lock().lines();
    let t: usize = lines.next().unwrap().unwrap().trim().parse().unwrap();
    for _ in 0..t {
        let n: usize = lines.next().unwrap().unwrap().trim().parse().unwrap();
        let vals_line = lines.next().unwrap().unwrap();
        if n == 0 {
            writeln!(out, "").unwrap();
            continue;
        }
        let toks: Vec<&str> = vals_line.split_ascii_whitespace().collect();
        let mut result: Vec<&str> = Vec::with_capacity(toks.len());
        let mut k: i64 = (toks.len() as i64) - 1;
        while k >= 0 {
            result.push(toks[k as usize]);
            k -= 1;
        }
        writeln!(out, "{}", result.join(" ")).unwrap();
    }
}
